<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>Parsers - a module that includes code for reading files in formats commonly used in proteomics</title>
<link rel="stylesheet" href="styles/perldoc.css" type="text/css" />
<meta http-equiv="content-type" content="text/html; charset=utf-8" />
<link rev="made" href="mailto:_postfix@neige.apple.com" />
</head>

<body>


<!-- INDEX BEGIN -->
<div name="index">
<p><a name="__index__"></a></p>

<ul>

	<li><a href="#name">NAME</a></li>
	<li><a href="#author">AUTHOR</a></li>
	<li><a href="#synopsis">SYNOPSIS</a></li>
	<li><a href="#description">DESCRIPTION</a></li>
	<li><a href="#methods">METHODS</a></li>
	<li><a href="#see_also">SEE ALSO</a></li>
</ul>

<hr name="index" />
</div>
<!-- INDEX END -->

<p>
</p>
<h1><a name="name">NAME</a></h1>
<p>Parsers - a module that includes code for reading files in formats commonly used in proteomics</p>
<p>
</p>
<hr />
<h1><a name="author">AUTHOR</a></h1>
<p>Dan MacLean (<a href="mailto:dan.maclean@tsl.ac.uk">dan.maclean@tsl.ac.uk</a>)</p>
<p>
</p>
<hr />
<h1><a name="synopsis">SYNOPSIS</a></h1>
<pre>
        use PhosphoTools::Parsers;</pre>
<pre>
        my $mgf_file = '/home/macleand/Desktop/my_recent_experiment.mgf';
        my @mgfs = Formatters::get_mgf($mgf_file);</pre>
<p>
</p>
<hr />
<h1><a name="description">DESCRIPTION</a></h1>
<p>The Parsers module contains code for reading in various TSL and proteomics formats. Extra formats can be added on request. The module does not create objects like some of the other modules, rather its methods are accessed in the manner of subroutines in external files.</p>
<p>
</p>
<hr />
<h1><a name="methods">METHODS</a></h1>
<dl>
<dt><strong><a name="get_motifs_from_motifx" class="item">get_motifs_from_motifx(html_file)</a></strong>

<dd>
<p>Will read in all the motifs in a web page generated by the motif-x peptide motif finder and return a hash that has as a key the text
representation of the mofif and as a value an array of peptides that contain that motif.</p>
</dd>
<dd>
<pre>
        my $html = '/home/macleand/Desktop/motifx_output.htm';
        my %motifs = Parsers::get_motifs_from_motifx($html);</pre>
</dd>
</li>
<dt><strong><a name="parse_phossites" class="item">parse_phossites(ptm)</a></strong>

<dd>
<p>When provided with a PTM score line in format:</p>
</dd>
<dd>
<pre>
        PEP[T]IDESEQ : 0.005 ; PEPTIDE[S]EQ : 0.0000001 ;</pre>
</dd>
<dd>
<p>will return a hash with keys that are the peptide sequence and values that are the score for the PTM. This method exists as a number of older
scripts and data files encode the PTM in this manner</p>
</dd>
<dd>
<pre>
        my $ptm_line = 'PEP[T]IDESEQ : 0.005 ; PEPTIDE[S]EQ : 0.0000001 ;';
        my %ptms = Parsers::parse_phossites($ptm_line);</pre>
</dd>
</li>
<dt><strong><a name="get_go_terms" class="item">get_go_terms(go_slim_file)</a></strong>

<dd>
<p>When provided with a TAIR style GO SLIM GO annotation file, will return a hash with key of protein accession and a value of a hash with keys 
of a string of TERM ID ASPECT seperated by ';'</p>
</dd>
<dd>
<pre>
        my $go_file = '/home/macleand/Desktop/TAIR9_GOSLIM';
        my %go_terms = Parsers::get_go_terms($go_file);</pre>
</dd>
</li>
<dt><strong><a name="get_pfamhits" class="item">get_pfamhits(pfamscan_file)</a></strong>

<dd>
<p>When provided with a pfamscan output file, will return a hash with key of protein accession and a value of a hash with keys 
of a string of DOMAIN_NAME START STOP P_VALUE seperated by ';'</p>
</dd>
<dd>
<pre>
        my $pfam_scan = '/home/macleand/Desktop/TAIR9_PFAMSCANS';
        my %pfams = Parsers::get_pfamhits($pfam_scan);</pre>
</dd>
</li>
<dt><strong><a name="get_proteins" class="item">get_proteins(protein_fasta_file)</a></strong>

<dd>
<p>When provided with a Protein sequence FASTA format file, will return a hash with key of protein accession and a value of a hash with keys 
'sequence' which has the protein sequence and 'description' which holds a description.</p>
</dd>
<dd>
<pre>
        my $proteins = '/home/macleand/Desktop/TAIR9.faa';
        my %proteins = Parsers::get_proteins($proteins);

</pre>
</dd>
</li>
<dt><strong><a name="get_mgf" class="item">get_mgf(mgf)</a></strong>

<dd>
<p>When provided with an MGF file, will return an array of hashes with key of each line of the mgf record</p>
</dd>
<dd>
<pre>
        my $mgf = '/home/macleand/Desktop/my_mgf.mgf';
        my @mgfs = Parsers::get_mgf($mgf);

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="see_also_phosphotools__formatters__bio__seqio_">SEE ALSO
PhosphoTools::Formatters; Bio::SeqIO;</a></h1>

</body>

</html>
